### DISCUSSION TITLE
where is the source code of drawing segm? i want to caculate area of segementation or mask.

### DISCUSSION THREAD

**Discussion by jwyng2000** [open-mmlab/mmdetection#7939] [Q&A] [answered]
hello im newcommer of ML DL Engeneer. I revised mask rcnnn webcam module to stream on local and network.

Now i have only little things to do on my project, and it is the warn someone when specific segmentation(CLASSES tuple 0)  
became specific size(bigger size).

I tryed to figure out or measure the size of segm, with debuging, finding in source codes and googling all weaks. 
But i couldn't find the solution.

So I temporarily added the code like `if '0' in model.CLASSES: print('warning') ` 
But it is not working in my intention

Can any one help me? 
Thank you!!

`
# Copyright (c) OpenMMLab. All rights reserved.
import argparse
import time

import cv2
import torch
import datetime
import logging

from mmdet.apis import inference_detector, init_detector

logger = logging.getLogger(__name__)
logger.setLevel(logging.ERROR)
formatter = logging.Formatter('%(asctime)s %(name)s %(levelname)s %(message)s')

streamhandler = logging.StreamHandler()
streamhandler.setFormatter(formatter)
logger.addHandler(streamhandler)

timedfilehandler = \
    logging.handlers.TimedRotatingFileHandler(filename='logs/logfile', when='midnight',
                                              interval=1, encoding='utf-8')
timedfilehandler.setFormatter(formatter)
timedfilehandler.suffix = "%Y%m%d"
logger.addHandler(timedfilehandler)

def detect():

    device = torch.device('cuda:0')

    model = init_detector('../configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco_1.py', '../checkpoints/epoch_72.pth', device=device)

    camera = cv2.VideoCapture('rtsp://root:1234qwer!@219.248.139.168/AVStream1_1')

    print('Press "Esc", "q" or "Q" to exit.')

    f_logtime = datetime.datetime.now()
    f_logtimes = f_logtime.strftime('%Y-%m-%d_%H-%M-%S')
    f_log=open('./f_log/'+f_logtimes+'.txt', 'w')
    f_log.close()

    while True:
        ret_val, img = camera.read()

        #입력된 프레임이 10의 배수일때 모델에 입력, 결과 산출.
        if (int(camera.get(1)) % 10 == 0):
            result = inference_detector(model, img)

            # 현재 시간
            pottime = datetime.datetime.now()
            pottimes = pottime.strftime('%Y-%m-%d_%H-%M-%S')

            ch = cv2.waitKey(1)

            if ch == 27 or ch == ord('q') or ch == ord('Q'):
                break
                logger.error('작동중지')

            else:
                # 스트리밍용 결과 프레임 저장
                frame = model.show_result(img, result, score_thr=0.5)

                # 다시 인코딩
                ret, buffer = cv2.imencode('.jpg', frame)

                #프레임을 바이트로 변환후 넘기기
                frame = buffer.tobytes()
                yield (b'--frame\r\n' b'Content-Type: image/jpeg\r\n\r\n' +
                   bytearray(frame) + b'\r\n')

                # 모델에 넣고 검출된 결과를 로컬에 시간별 저장.
                model.show_result(
                    img, result, score_thr=0.5, out_file='detect_images/' + pottimes + '.jpg')

                # 모델에 물(0) 이 있나를 판별해주고 경고 해주기.
                if '0' in model.CLASSES:
                    print("Warning")

                # 생성로그 / 파일에 생성된 파일목록 저장
                f_log = open('./f_log/' + f_logtimes + '.txt', 'a')
                f_log.write(pottimes+'.jpg  ')
                f_log.close()

                print('good')

`

**Comment by jwyng2000** [comment, marked as answer]
[mmdetection/mmdet/core/mask/structures.py](https://github.com/open-mmlab/mmdetection/blob/ae453fa92ffebcbd224b72f6d48e0b8699424450/mmdet/core/mask/structures.py#L301-L303)

Lines 301 to 303 in [ae453fa](https://github.com/open-mmlab/mmdetection/commit/ae453fa92ffebcbd224b72f6d48e0b8699424450)

guys i found this.

But still don't know how to use it..

 def areas(self): 
     """See :py:attr:`BaseInstanceMasks.areas`.""" 
     return self.masks.sum((1, 2))

**Comment by AristotleWang** [comment]
Did you solve this problem? I also want to develop similar functional modules

**Comment by jwyng2000** [comment]
yeah i found some matters. i will send you if you want to see. when i go to
my office to work. today is sunday :)

2022년 10월 18일 (화) 오후 5:52, AristotleWang ***@***.***>님이 작성:

**Reply by AristotleWang** [reply]
Thanks for your reply.Could you please send me the code when it is convenient？

**Reply by jwyng2000** [reply]
frame = model.show_result(img, result, score_thr=0.7)
                logger.debug("nice")

                # 바운딩 박스가 출현시
                if len(result[0][0]) != 0:
                    # print("리설트값00")
                    # print(result[0][0])
                    # 마스크값 저장
                    mask_imsi = result[1][0][0]
                    # print("마스크 출력: ", mask_imsi)
                    # 마스크값에서 True인 부분 추출및 저장
                    maskArea_pixel = mask_imsi[mask_imsi > 0].shape
                    maskArea_pixel = maskArea_pixel[0]
                    print("마스크 픽셀의 합계(면적) : ", maskArea_pixel)

                    # 프레임의 bbox위에 면적합을 띄우기
                    cv2.putText(frame, "Areas: {}".format(maskArea_pixel),
                                (50, 50), cv2.FONT_HERSHEY_PLAIN, 2, (255, 0, 0), 2)
